microsoft/PowerToys
# Upgrade check-spelling to 0.0.17-alpha

## Summary of the Pull Request

**What is this about:**

In the past, plurals `foo`+`s` and past tenses `foo`+`ed` were
automatically tolerated. This turned out to be a bad design choice on my
part.

The basic example is that `potatos` would sometimes be treated as a
mistake and sometimes not (depending on the presence of `potato`).

You can see in this PR, that this logic resulted in `SNAPED` being
accepted as a word along with `actioned` -- there's nothing
intrinsically wrong w/ the latter, but unfortunately in order to screen
out the former, my shortcut just couldn't stick around. This means that
the `expect` files will grow perhaps by a tiny bit, but as
you can see, not really by much.

When GitHub initially introduced GitHub Actions, the event for
`pull_request` was created without enough permission for a tool like
this to work properly. I worked around that by using the `schedule`
event. In 2020, they introduced a replacement event
`pull_request_target` which has enough permission. This means that I can
stop relying on the `schedule` event.
-- PRs from forks will now have ❌ when there are unrecognized words (previously this only appeared for local branches).

**What is include in the PR:** 

- two spelling fixes
- pattern improvements
- upgraded spell check action

**How does someone test / validate:** 

You can pull this branch into your own fork and make commits on top of it, or make pull requests into it.

I have a temporary repository https://github.com/check-spelling/PowerToys/ which is running this change that can receive PRs (I will delete the repo at some point).

New misspelled words will still result in a comment.

## Quality Checklist


## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
